dynamics_controllers/README.md

```diff
@@ -0,0 +1,158 @@
+  joint1/pid: {p: 10.0, i: 0.0, d: 5.0}
+  joint2/pid: {p: 10.0, i: 0.0, d: 5.0}
+```
+
+To use the new controllers, you can change it, *e.g.*, as follows:
+```yaml
+gravity: [0, 0, -9.81] # this should be projected on the base frame
+
+position_controller:
+  type: dynamics_controllers/KdlChainController
+  sub_controller:
+    # the subcontroller is basically the controller you had before!
+    type: effort_controllers/JointGroupPositionController
+    joints:
+      - joint1
+      - joint2
+    joint1/pid: {p: 10.0, i: 0.0, d: 5.0}
+    joint2/pid: {p: 10.0, i: 0.0, d: 5.0}
+
+velocity_controller:
+  type: dynamics_controllers/KdlTreeController
+  sub_controller:
+    # the subcontroller is basically the controller you had before!
+    type: effort_controllers/JointGroupVelocityController
+    joints:
+      - joint1
+      - joint2
+    joint1/pid: {p: 10.0, i: 0.0, d: 5.0}
+    joint2/pid: {p: 10.0, i: 0.0, d: 5.0}
+```
+
+Note that you will likely have to tune again the gains of the sub-controllers to
+achieve good performances.
+
+#### Detailed Controllers Description
+
+Below, parameters relative to each controller are listed.
+Those marked as "searched for" can live in any parent namespace since the
+`NodeHandle::searchParam` method is used to locate them.
+This facilitates "sharing" common parameters such as the gravity vector, which
+should not depend on a specific controller (at least in principle!).
+
+##### KdlChainController
+
+- `sub_controller`: should contain the configuration of a controller of type
+  compatible with `EffortJointInterface`.
+- `robot_description`: should contain the URDF of the robot. Searched for.
+- `gravity`: list with three elements, representing the gravity vector in the
+  base frame of the chain. Searched for. Default: `[0,0,0]`.
+- `chain_base`: base of the chain. Searched for. Default: name of the root link
+  from the robot description.
+- `chain_tip`: tip of the chain. Searched for. Default: if a single "branch"
+  is rooted at `base_link`, `chain_tip` will correspond to the last link of the
+  chain. If at any point multiple children are found, loading will fail.
+- `gravity_compensation_only`: as discussed above. Default: `false`.
+
+##### KdlTreeController
+
+- `sub_controller`: should contain the configuration of a controller of type
+  compatible with `EffortJointInterface`.
+- `robot_description`: should contain the URDF of the robot. Searched for.
+- `gravity`: list with three elements, representing the gravity vector in the
+  base frame of the chain. Searched for. Default: `[0,0,0]`.
+- `tree_root`: base of the tree. Searched for. Default: name of the root link
+  from the robot description.
+- `gravity_compensation_only`: as discussed above. Default: `false`.
+
+
+
+### Limitations
+
+- They all assume a static base. The main implication is that you cannot
+  "serially join" controllers for different parts of the robot.
+- They require proper identification of the Dynamic Parameters.
```

Comment:
Could you please add some pointers on how one would go about this?

Reply:
That is supposed to be a warning about the fact that model-based computed torque controls require a sufficiently good identification of the parameters. Do you want me to add references about possible identification strategies?